### PR TITLE
Removing {fs} dependency (Fixes #69)

### LINKED ISSUE #69: Remove fs dependency
My understanding is there is no major reason to keep fs so let's remove it to keep dependencies low.

### PR DESCRIPTION
**Pull request by Ilia-Kosenkov**
Removes {fs} in favor of base R (with some minor limitations). Fixes #69.

### CODE REVIEW

File: R/track_rust_source.R [hunk 1]
```diff
@@ -2,15 +2,58 @@
...
 pretty_rel_path <- function(path, search_from = ".") {
-  fs::path_rel(path, start = rprojroot::find_package_root_file(path = search_from))
+  # Absolute path to the package root
+  package_root <- normalizePath(
+    rprojroot::find_package_root_file(path = search_from),
+    winslash = "/"
+  )
+
+  # Absolute path.
+  # `path` may not exists, so `mustWork` suppresses unnecessary warnings.
```
Comment (clauswilke) [inline R/track_rust_source.R:17]: "may not *exist*"

File: R/track_rust_source.R [hunk 2]
```diff
@@ -111,3 +155,17 @@ needs_compilation <- function(path = ".", quiet = FALSE) {
   TRUE
 }
 
+# Replaces `fs::file_info`.
```
Comment (clauswilke) [inline R/track_rust_source.R:158]: Instead of "Replaces", write "Equivalent to"?

**Review by clauswilke — APPROVED** [review]
Thanks!

**Comment by clauswilke** [comment]
Two more comments:

1. I assume you've seen #79. It would be best to address this here.
2. This brings up the issue that for any non-trivial helper functions that we write, we should add some unit test functions that verify the function behavior. Here, for `pretty_rel_path()`, having a unit test that tests the relative path in a package and a unit test that tests the behavior outside of a package would be great. Similarly, one or two unit tests for `get_file_info()` would be useful too.

**Comment by Ilia-Kosenkov** [comment]
@clauswilke,
I am not sure how to test `get_file_info` -- it basically calls `file.info` and wraps the output. 
To be able to *really* test it, we need to create a file with specific `ctime`/`mtime`/`atime`, which then can be read and its properties compared to the expected values. I do not see how we can achieve this with R without external dependencies.

The only base tool I found is [`Sys.setFileTime`](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/Sys.setFileTime), which modified `mtime`. Perhaps it is enough for us, as `mtime` is the only field we really care about.

**Comment by clauswilke** [comment]
I think you're overthinking this. `get_file_info()` is a wrapper and convenience function, so you test that functionality. Creat a temp, run both `get_file_info()` and `file.info()` on it, and test that the result is the same. Also test that the output has the format you expect, and that `get_file_info()` behaves reasonably on bad input, e.g. on a file that doesn't exist.

File: R/track_rust_source.R [hunk 3]
```diff
@@ -1,16 +1,89 @@
...
+        pattern = package_root,
+        case_insensitive = TRUE
+      )
+  ) {
+    return(path)
+  }
+
+  # If `path` is a subpath of `package_root`,
+  # then `path` contains `package_root` as a substring.
+  # This removes `pacakge_root` substring from `path`,
```
Comment (clauswilke) [inline R/track_rust_source.R:62]: spelling: `package_root`

**Review by clauswilke — APPROVED** [review]
Thanks, looks good. I found one typo, which I pointed out in a separate comment.

File: tests/testthat/test-helper-functions.R [hunk 4]
```diff
@@ -94,3 +94,53 @@ test_that("`pretty_rel_path()` works", {
     "A/B/C/D.F"
   )
 })
+
+
+# Verifies that `rextendr::get_file_info()` returns the same as base R
+# `file.info()` for files and directories.
+test_that("`get_file_info()` and `file.info()` outptus are equivalent", {
```
Comment (clauswilke) [inline tests/testthat/test-helper-functions.R:101]: typo: "output"

File: tests/testthat/test-helper-functions.R [hunk 5]
```diff
@@ -94,3 +94,53 @@ test_that("`pretty_rel_path()` works", {
...
+  n_test_files <- 10L
+  n_test_dirs <- 2L
+  # Creates 10 temp files in 10 *different* temp directories,
```
Comment (clauswilke) [inline tests/testthat/test-helper-functions.R:104]: Do we need 10? Maybe 3 is sufficient?

**Comment by Ilia-Kosenkov** [comment]
Unless I made another typo somewhere, I think everything has been implemented & tested.

**Comment by clauswilke** [comment]
Yes, please go ahead and merge. Thanks!

**Pull request merged**